static-site/src/components/ListResources/Showcase.tsx

```diff
@@ -9,26 +10,66 @@ import { LIST_ANCHOR } from "./index";
 import { Card, FilterOption, Group } from './types';
 
 const cardWidthHeight = 160; // pixels
+const expandPreviewHeight = 50 //pixels
 
 type ShowcaseProps = {
     cards: Card[]
     setSelectedFilterOptions: React.Dispatch<React.SetStateAction<readonly FilterOption[]>>
 }
 
 export const Showcase = ({cards, setSelectedFilterOptions}: ShowcaseProps) => {
-  if (!cards.length) return null;
+
+  const [numRows, setNumRows] = useState<number>(0);
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+
+  const toggleExpand = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  /**
+   * Function that runs on changes to the container.
+   * Used to compute the number of rows of cards upon resize.
+   */
+  function cardsContainerRef(cardsContainer: HTMLDivElement) {
+    if (!cardsContainer) return;
+
+    const cards = cardsContainer.children;
+
+    if (cards && cards.length > 0) {
+      const leftBorder = (cards[0] as HTMLElement).offsetLeft;
+
+      let newNumRows = 0;
+      for (let i = 0; i < cards.length; i++) {
+        if ((cards[i] as HTMLElement).offsetLeft == leftBorder) {
+          newNumRows++;
+        }
+      }
+      if (numRows != newNumRows) {
+        setNumRows(newNumRows);
+      }
+    }
+  }
```

Comment:
_non-blocking_

Why do these layout calculations ourselves instead of using flexbox? I'd think it'd be simpler to implement and maintain with flexbox.

Reply:
I did it this way following recommendations from online sources. Not sure what you mean by "using flexbox", but your comment realized that the calculations were unnecessary and I've simplified it in d60cf39011a03479c40fdbc442d030e22c4a4af3.